sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -87,16 +87,20 @@ public void openConnectionIfNeeded() {
             return;
         }
 
+        Uri addressUri = this.addressUris.stream().findFirst().get();
```

Comment:
Is it ever possible that this.addressUris is empty at this point?

Reply:
That would throw above

Reply:
It is unlikely. The SDK adds to `addressUris` each time a connection is opened (open connections flow) or the endpoint receives a request. But having a presence check is good practice.